storage/cache/map_subtree_cache.go

```diff
@@ -61,7 +61,7 @@ func populateMapSubtreeNodes(treeID int64, hasher hashers.MapHasher) storage.Pop
 				return nil, nil
 			},
 			func(depth int, index *big.Int, h []byte) error {
-				if depth == 0 {
+				if depth == 0 && len(st.Prefix) > 0 {
```

Comment:
This looks like an analogue to the log case. You should test this change with a map large enough to span several subtrees as the unit tests won't be enough. Possibly the map integration test is enough but I don't know how robust it is.